cumulusci/robotframework/Salesforce.py

```diff
@@ -488,10 +490,93 @@ def salesforce_insert(self, obj_name, **kwargs):
         """ Inserts a Salesforce object setting fields using kwargs and returns the id """
         self.builtin.log("Inserting {} with values {}".format(obj_name, kwargs))
         obj_class = getattr(self.cumulusci.sf, obj_name)
+
         res = obj_class.create(kwargs)
         self.store_session_record(obj_name, res["id"])
+
         return res["id"]
 
+    def salesforce_init_object(self, obj_name, **fields):
+        obj = {"attributes": {"type": obj_name}}  # Object type to create
+        obj.update(fields)
+        return obj
+
+    def salesforce_init_objects(self, obj_name, number_to_create, **fields):
```

Comment:
See the quip doc [Creating test data in robot tests](https://salesforce.quip.com/jCyVA1a0DT9i) for my comments about this keyword as a whole.